Python/wrfrun/unCMIP6.py

```diff
@@ -0,0 +1,622 @@
+                                            ndatefound2 = ndatefound2+1
+                                if ndatefound2!=1:
+                                    raise ValueError('Error: Could not find the file with the input data.')
+                self.filestrdates.append(filestartdate.strftime(dformat)+'-'+fileenddate.strftime(dformat))
+            else:
+                self.filestrdates.append('')
+        
+        # plev and lat and lon variables
+        self.plev = np.array(None)
+        self.lat = np.array(None)
+        self.lon = np.array(None)
+
+        # Assemble lats, lons, and the data
+        print('Loading data ...')
+        c = 0 # Counter.
+        for idx, itm in self.vtable.iterrows():
+            # Var name and other info
+            varname = itm['src_v']
+            outvarname = itm['aim_v']
+            lvltype = itm['type']
+            lvlmark = itm['lvlmark']
+            frq = itm['freq']            
+            if lvlmark == 'None':
+                lvlmark = ''
+            # File name
+            fn = input_root+'/'+varname+'_'+frq+lvlmark+'_'+self.model_name
+            fn = fn+'_'+self.exp_id+'_'+self.esm_flag+'_'+self.grid_flag
+            if self.filestrdates[c]!='':
+                fn = fn+'_'+self.filestrdates[c]+'.nc'
+            else:    
+                fn = fn+'.nc'  
+            # Open dataset
+            ds = xr.open_dataset(fn)
+            # lat and lon handeling
+            if ((self.lat==None).all() and (self.lon==None).all()):
+                if ((varname=='tos') or (varname=='siconc') or (varname=='sithick')):
+                    raise ValueError("The first field in the table should be a variable different than tos, siconc or sithick.")
+                self.lat = np.array(ds.lat.values)
+                self.lon = np.array(ds.lon.values)
+            else:
+                if ((varname!='tos') and (varname!='sithick') and (varname!='siconc')):
+                    if not((np.allclose(np.array(ds.lat.values),self.lat,rtol=0.0,atol=1.0e-12)) and \
+                        (np.allclose(np.array(ds.lon.values),self.lon,rtol=0.0,atol=1.0e-12))):
+                        raise ValueError("Error: Inconsistent lat/lon values between different fields.")
+            # plev handeling for 3D fields
+            if lvltype=='3d':
+                if ((self.plev==None).all()):
+                    if (varname=='ua') or (varname=='va'):
+                        raise ValueError("The first 3D field in the table should be a variable different than ua and va.")
+                    self.plev = np.array(ds.plev.values)
+                else:
+                    if not(np.allclose(np.array(ds.plev.values),self.plev,rtol=0.0,atol=1.0e-12)):
+                        raise ValueError("Error: Inconsistent pressure levels between 3D fields.")            
+            # Find the date to read
+            if (self.filestrdates[c]!=''):
+                avail_times_t = ds[varname].time.values
```

Comment:
You are opening these files with xarray and CMIP6 should be CF compliant, so it should be able to parse the time axis and load as datetime64; then you can just slice the closest date with `ds.sel(time=date, method="nearest")`

Reply:
https://docs.xarray.dev/en/stable/user-guide/indexing.html#nearest-neighbor-lookups

Reply:
We talked about this. This whole code part is now gone.